design/fips-compliance.md

```diff
@@ -0,0 +1,259 @@
+and find the errors. Furthermore, `kubernetes-conjur-demo` runs also tests for
+the `secretless-broker` so failures there will fail builds of the authn-client.
+
+My suggestion is as follows:
+  - If we can get 10 days for testing then we should research and design the best 
+    option to run the integration tests in the same repo. Finding a good solution 
+    will be useful also in the `secrets-provider-for-k8s` and in the `secretless-broker`.
+    The options are:
+      - `secrets-provider` style: deploy different scenarios and check output   
+      - [`KInD`](https://github.com/kubernetes-sigs/kind)
+      - Any other option found in the research
+  - If we can't get 10 days, implement option (b). We won't have the same confidence 
+    and we may merge to `master` code that will break the integration tests but the
+    implementation time will be very short. Developers who merge to `master` will need
+    to verify that the build `kubernetes-conjur-demo` succeeded after the merge
+    (and in case of failure we will get notified in Slack)
+
+### Integration Tests
+
+Regardless of how we will run our tests, it is not optimal that we have only
+ a vanilla flow. We should add another test where in case the authenticator
+ -client fails to authenticate with Conjur we don't provide an access token
+  and the log shows `CAKC015E Login failed`.
+  
+We do not need to test different permutations of error flows (e.g host does
+not exist, host is not permitted on the authn-k8s/prod authenticator) as
+these test run in the `conjur` repository. As far as the authenticator-client
+ is concerned, the output is binary - Authentication success or failure.
+   
+| **Scenario**            | **Given**                                                    | **When**                                                 | **Then**                                                                                               |
+|-------------------------|--------------------------------------------------------------|----------------------------------------------------------|--------------------------------------------------------------------------------------------------------|
+| Authentication succeeds | A Running Conjur cluster with a configured K8s Authenticator | I run the authenticator client with a valid k8s host     | An access token is provided to the application container and it can retrieve a secret with it          |
+| Authentication fails    | A Running Conjur cluster with a configured K8s Authenticator | I run the authenticator client with a non-valid k8s host | An access token is not provided to the application container and the log shows `CAKC015E Login failed` |
+
+## Docs
+
+We should document any change that will affect the customer (e.g if we release 
+both FIPS & Non-FIPS versions).
+
+If no customer-facing changes are introduced, we will not need to add documentation.
```

Comment:
From meeting:
Sigal opinion ** Add here a note saying that the proposed solution does not have customer-facing changes